src/ui/dialog-vkbasalt.ui

```diff
@@ -0,0 +1,819 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!-- Created with Cambalache 0.10.2 -->
+<interface>
+  <!-- interface-name dialog-vkbasalt.ui -->
+  <requires lib="gtk" version="4.0"/>
+  <requires lib="libadwaita" version="1.0"/>
+  <template class="VkBasaltDialog" parent="AdwWindow">
+    <property name="default-width">550</property>
+    <property name="modal">True</property>
+    <property name="title" translatable="yes">vkBasalt Settings</property>
+    <child>
+      <object class="GtkBox">
+        <property name="orientation">vertical</property>
+        <child>
+          <object class="AdwHeaderBar">
+            <property name="show-end-title-buttons">False</property>
+            <child type="start">
+              <object class="GtkBox">
+                <property name="spacing">6</property>
+                <child>
+                  <object class="GtkButton" id="btn_cancel">
+                    <property name="label" translatable="yes">_Cancel</property>
+                    <property name="use_underline">True</property>
+                    <property name="action_name">window.close</property>
+                  </object>
+                </child>
+                <child>
+                  <object class="GtkShortcutController">
```

Comment:
This has to be a direct child of the AdwWindow, otherwise it wont close the window if the focus is not in the GtkBox which is inside the headerbar.